lib/hmmlearn/_hmmc.pyx

```diff
@@ -64,12 +69,79 @@ def _forward(int n_samples, int n_components,
 
                 fwdlattice[t, j] = _logsumexp(work_buffer) + framelogprob[t, j]
 
+def _forward_scaling(int n_samples, int n_components,
+                     dtype_t[:] startprob,
+                     dtype_t[:, :] transmat,
+                     dtype_t[:, :] frameprob,
+                     dtype_t[:, :] fwdlattice,
+                     dtype_t[:] scaling_factors,
+                     dtype_t min_scaling=1e-90):
```

Comment:
Is there a specific reason for this value?  If you really want to introduce a check (perhaps just allowing underflow is fine), it would make sense to allow going to the limit of double resolution, i.e. 1e-300.

Reply:
I'll update the value, and I have a test I can add to demonstrate how underflow can occur with the scaling algorithm.